addons/base_vat/models/res_partner.py

```diff
@@ -116,42 +140,70 @@ def fix_eu_vat_number(self, country_id, vat):
                 vat = country_code + vat
         return vat
 
-    @api.constrains('vat')
-    def check_vat(self):
-        if self.env.context.get('company_id'):
-            company = self.env['res.company'].browse(self.env.context['company_id'])
-        else:
-            company = self.env.user.company_id
-        if company.vat_check_vies:
-            # force full VIES online check
-            check_func = self.vies_vat_check
-        else:
-            # quick and partial off-line checksum validation
-            check_func = self.simple_vat_check
+    @api.depends('vat', 'commercial_partner_id.country_id')
+    def _compute_base_vat_vies_check_status(self):
         for partner in self:
-            if not partner.vat:
-                continue
-            #check with country code as prefix of the TIN
-            vat_country, vat_number = self._split_vat(partner.vat)
-            if not check_func(vat_country, vat_number):
-                #if fails, check with country code from country
-                country_code = partner.commercial_partner_id.country_id.code
-                if country_code:
-                    if not check_func(country_code.lower(), partner.vat):
-                        msg = partner._construct_constraint_msg(country_code.lower())
-                        raise ValidationError(msg)
-
-    def _construct_constraint_msg(self, country_code):
-        self.ensure_one()
-        vat_no = "'CC##' (CC=Country Code, ##=VAT Number)"
-        vat_no = _ref_vat.get(country_code) or vat_no
-        if self.env.context.get('company_id'):
-            company = self.env['res.company'].browse(self.env.context['company_id'])
-        else:
+            company = self.env.context.get('company_id')
+            if not company:
+                company = self.env.user.company_id
+
+            check_rslt = None
+            if partner.vat:
+                if partner.parent_id and partner.vat == partner.parent_id.vat:
+                    partner.base_vat_vies_check_status = partner.parent_id.base_vat_vies_check_status
+                    continue
+                elif company.vat_check_vies:
+                    check_rslt = partner._check_vat(self.vies_vat_check)
+
+            if check_rslt == None:
+                partner.base_vat_vies_check_status = 'unknown'
+            else:
+                partner.base_vat_vies_check_status = check_rslt and 'verified' or 'wrong'
+
+    @api.constrains('vat', 'commercial_partner_id')
```

Comment:
do we still need this constraint? I though we don't, since we introduced the onchange+warning

=> probably to remove